### PR TITLE
Add login function 

### PR DESCRIPTION
**Pull request by inquisitiv-e**
Completed:
- Used Meteor's AccountsUIWrapper following the tutorial [here](https://www.meteor.com/tutorials/react/adding-user-accounts)
`meteor add accounts-ui accounts-password`
- Currently, it shows ShoppingList Icon, AddDeal and Favorite Button (for adding to shopping list) only when a user logs in successfully. 

- No backend set-up yet. Thinking of data-structure for users to be something like:

`let newUser = {
      name --> username,
      shoppingList --> array of ObjectIDs of card
    };`

Before Login:
Screen Shot 2019-07-12 at 9 14 25 PM

After Login:
Screen Shot 2019-07-12 at 9 16 45 PM

To-do:

- [x] Refresh automatically on log-in

### CODE REVIEW

**Comment by inquisitiv-e** [comment]
> Instead of refreshing after user logs in. There can be a current user state that gets updated that will update the website, that way the updated state will trigger a rerender. Or you can use forceUpdate() as well somewhere.

**Comment by YaoLiu93** [comment]
We can customize the login ui, say user provides a first + last name.
https://stackoverflow.com/questions/15025621/how-to-style-meteor-js-loginbuttons

File: imports/ui/components/Card.jsx [hunk 1]
```diff
@@ -128,6 +110,25 @@ class CardComponent extends Component {
               </div>
             )}
           </div>
+          {/* Favorite icon is a part of the stretch goal to add wishlist, use later */}
+          {this.props.currentUser ? (
```
Comment (YaoLiu93) [inline imports/ui/components/Card.jsx]: Change to if statement

File: imports/ui/components/Modal.jsx [hunk 2]
```diff
@@ -39,15 +39,19 @@ class Modal extends Component {
...
-        <Fab
-          color="secondary"
-          aria-label="Edit"
-          className="NewPostFab"
-          onClick={this.handleOpen}
-          variant="extended"
-        >
-          <AddIcon /> Fresh Deal
-        </Fab>
+        {this.props.currentUser ? (
```
Comment (YaoLiu93) [inline imports/ui/components/Modal.jsx]: Change to if statement.

File: imports/ui/components/ShoppingListButton.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class ShoppingListButton extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <div>{this.props.currentUser ?
```
Comment (YaoLiu93) [inline imports/ui/components/ShoppingListButton.jsx]: change to if statement. 
Fix formatting.

**Comment by nicolehli** [comment]
> No backend set-up yet. Thinking of data-structure for users to be something like:
> let newUser = { name --> username, shoppingList --> array of ObjectIDs of card };

{
	username: email,
	cart: [ _id, _id…]
}

**Pull request merged**